assets/js/components/KeyMetrics/ConfirmSitePurposeChangeModal.js

```diff
@@ -161,25 +156,37 @@ function ConfirmSitePurposeChangeModal( {
 		);
 		await saveUserInputSettings();
 
-		setIsSaving( false );
-
-		// Handle internal tracking.
-		trackEvent(
-			`${ viewContext }_kmw-settings-tailored-metrics-suggestions`,
-			'confirm_update_metrics_selection',
-			'conversion_reporting'
-		);
-
 		onClose();
 	}, [
 		saveUserInputSettings,
 		onClose,
 		setIsSaving,
 		setUserInputSetting,
 		userInputPurposeConversionEvents,
-		viewContext,
 	] );
 
+	const prevDialogActive = usePrevious( dialogActive );
+
+	useEffect( () => {
+		if ( prevDialogActive === true && dialogActive === false ) {
+			if ( isSaving ) {
+				// Handle internal tracking.
```

Comment:
We can improve the comment for a better clarity, or remove it

Something like:
```suggestion
				// Handle internal tracking when confirmation CTA is clicked.
```

Reply:
Thanks @zutigrm addressed in my latest commit.